samples/drivers/adc/sample.yaml

```diff
@@ -5,7 +5,8 @@ tests:
     tags: ADC
     depends_on: adc
     platform_allow: nucleo_l073rz disco_l475_iot1 cc3220sf_launchxl
-        cc3235sf_launchxl stm32l496g_disco
+        cc3235sf_launchxl stm32l496g_disco gd32f350r_eval gd32f450i_eval
+        gd32vf103v_eval
```

Comment:
Why not enable F403? Is there any particular reason?

Reply:
No GD32F403 evaluation board on my side.

adc-gd32 is a general GD32 ADC driver. So, I think it works on GD32F403 too.

Reply:
There have two remain issue for gd32f403 ADC.
1. pinconfig for GD32F403Zx in hal_gigadevice missing PC3. (To run the ADC sample on GD32F403Z-EVAL board)
2. ADC0 and ADC1 shared interrupt 18, but ADC2 use an individual interrupt 47.

So, after fix the PC3 missing issue, ADC0 and ADC1 will work well. But ADC2 looks need more work to adjust adc_gd32 driver.

Reply:
gd32f403z_eval also support the sample `samples/drivers/adc`